### PR TITLE
upgrade to Typescript 3.2

### PR DESCRIPTION
**Pull request by shiftkey**
## Overview

VSCode is now using TS 3.2 by default, so you might encounter errors if you haven't forced VSCode to use the workspace version. I looked into what had changed between TS 3.1 and 3.2 and this was the minimum work to get things compiling again - looks like some changes to how type inference works with `Pick`.

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/lib/stores/app-store.ts [hunk 1]
```diff
@@ -762,11 +762,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
...
+      const newState: IDisplayHistory = {
+        kind: HistoryTabMode.History,
+      }
+
       this.repositoryStateCache.updateCompareState(repository, () => ({
         tip: currentSha,
-        formState: {
-          kind: HistoryTabMode.History,
-        },
+        formState: newState,
```
Comment (j-f1) [inline app/src/lib/stores/app-store.ts]: Why these changes? What kind of errors were thrown?

Comment (shiftkey) [reply app/src/lib/stores/app-store.ts]: This change is because the type inference couldn't verify this satisfies the criteria of `formState`:

> Type '{ kind: HistoryTabMode; }' is not assignable to type 'IDisplayHistory | ICompareBranch'.

If I break it out it becomes clearer where the problem lies:



The next time we're in making changes to the compare tab we can revisit these abstractions, but for now I just want to appease the Type System Gods™
